xbmc/interfaces/json-rpc/SettingsOperations.cpp

```diff
@@ -684,6 +695,26 @@ bool CSettingsOperations::SerializeSettingAddon(const CSettingAddon* setting, CV
   return true;
 }
 
+bool CSettingsOperations::SerializeSettingDate(const CSettingDate* setting, CVariant &obj)
```

Comment:
You mean add the JSON serialization to the `CSettingFoo` classes? Right now those classes only have deserialization logic to read the setting definition from XML. This would be serializing the setting definition to JSON. So it might be a bit confusing.

Furthermore any changes to the JSON serialization implementation could require changes to the JSON-RPC API schema. That is more obvious if the serialization code is located with the rest of the JSON-RPC API code.

But I agree that it's not the cleanest solution.